src/journal/journald-context.c

```diff
@@ -268,6 +271,51 @@ static int client_context_read_label(
         return 0;
 }
 
+static int client_context_read_log_regex(
+                const char *cgroup,
+                const char *key,
+                pcre2_code **regex) {
+        _cleanup_free_ char *pattern = NULL;
+        pcre2_code *new_regex = NULL;
+        int r;
+
+        assert(cgroup);
+
+        r = cg_get_xattr_malloc(SYSTEMD_CGROUP_CONTROLLER, cgroup, key, &pattern);
+        if (r == -ENODATA)
+                /* -ENODATA should not be considered as an error. It means the key
+                 * is not defined. */
+                return 0;
+        if (r < 0)
+                return log_debug_errno(r, "Failed to get %s xattr for %s: %m", key, cgroup);
+
+        r = dlopen_pcre2();
+        if (r < 0)
+                return r;
+
+        r = pattern_compile_and_log(pattern, 0, &new_regex);
+        if (r < 0)
+                return r;
+
+        free_and_replace_full(*regex, new_regex, pattern_free);
+
+        return 0;
```

Comment:
Can be simplified to `return free_and_replace_full(*regex, new_regex, pattern_free);`